internal/stack/resources.go

```diff
@@ -152,9 +152,25 @@ func applyResources(profile *profile.Profile, stackVersion string) error {
 		"geoip_dir":            profile.Config(configGeoIPDir, "./ingest-geoip"),
 		"logstash_enabled":     profile.Config(configLogstashEnabled, "false"),
 		"self_monitor_enabled": profile.Config(configSelfMonitorEnabled, "false"),
-	})
+	}
+
+	var agentPorts []string
+	if err := profile.Unmarshal("stack.agent.ports", &agentPorts); err != nil {
+		return fmt.Errorf("failed to unmarshal stack.agent.ports: %w", err)
+	}
 
-	os.MkdirAll(stackDir, 0755)
+	if len(agentPorts) > 0 {
+		facter["agent_has_publish_ports"] = "true"
+		facter["agent_publish_ports"] = fmt.Sprintf("[%s]", strings.Join(agentPorts, ","))
+	} else {
+		facter["agent_has_publish_ports"] = "false"
+	}
+
+	resourceManager.AddFacter(facter)
+
+	if err := os.MkdirAll(stackDir, 0755); err != nil {
+		return fmt.Errorf("failed to create stack directory: %w", err)
+	}
```

Comment:
This should not be needed :thinking:

Reply:
I just added the err, `os.MkdirAll...` was already there but without being checked. If we want the error to be ignored there then I would prefer have it as `_ = os.MkdirAll...` so it is evident that we explicitly ignore this 🙂

Reply:
It is fine.